kqueen_ui/blueprints/manager/views.py

```diff
@@ -121,6 +120,9 @@ def handle(self, page):
             )
         except Exception as e:
             return handle_exception_for_ajax(e)
+        if clusters is None:
+            return jsonify({'session_expired': True})
```

Comment:
Here log in screen will appear. How do you suggest it handle it in kqueen_request? We rely on clusters data below in this function so we have to check the result anyway, I guess

Reply:
I still don't understand why this check is needed. We check for a valid session in login_required and superadmin_required

Reply:
Because everything in this method will turn into a string handled by JS eventually. So redirects or flash messages don't work here. We can only send a different response to JS.